repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/core/proxy/MisbehavingOriginTest.groovy

```diff
@@ -51,15 +51,15 @@ class MisbehavingOriginTest extends ReposeValveTest {
     }
 
     //TODO: I want this to return a 502, but I cannot, because repose internals can't deal with other exceptions
-    def "returns a 500 when the origin service doesn't respond with proper http"() {
+    def "returns a 503 when the origin service doesn't respond with proper http"() {
         given: "something is going to do bad things"
 
         when: "Request goes through repose"
         MessageChain mc = deproxy.makeRequest(url: reposeEndpoint)
 
-        then: "repose should return a 500 Internal Server Error"
+        then: "repose should return a 503 Service Unavailable"
         mc.receivedResponse != null
-        mc.receivedResponse.code == "500"
+        mc.receivedResponse.code == "503"
```

Comment:
We explicitly handle the `IOException` that is thrown as a result of the server closing the socket unexpectedly by setting the response code to `503`. I'm not sure why we were seeing `500` before, but is my expectation looking at the code. Although, I do wonder if this test should wait to close the server socket until the end of the test rather than immediately after writing some bytes on the wire. That would probably change the behavior in Repose... but that is neither here nor there.

Reply:
Looking at this again, it appears that the `500` is expected when an `IOException` isn't being thrown. My theory is that we hit that case if Apache HttpClient processes the garbage response before the test closes the connection, otherwise the closed connection leads to an `IOException` which results in a `503` instead of a `500`.

Reply:
Whelp. Waiting to close the socket leads to Apache HttpClient throwing a `SocketTimeoutException` instead of a `ConnectionResetException`. Both are `IOException`s, though, and should result in a `503`. So much for the theory that HttpClient was handling the response leading to `500`s instead. So I'm still not sure where the `500` or `200` are coming from.

Reply:
So yeah. When run on Jenkins, the log shows that a `org.apache.http.client.ClientProtocolException` is thrown, leading to the expected `500` response code. When run locally, a `SocketException` (Connection Reset) is thrown leading to the expected `503`. That is when checking out the exact same commit, cleaning with both Git and Gradle, and running the same single test.

At this point, I suspect it may be a JDK thing. I'm running `Java 1.8.0_161 Hotspot` while the Jenkins server is running `Java 1.8.0_131 OpenJDK`.

I'm just going to try and get it to pass on Jenkins.

Reply:
The `200` is from the `PowerFilterChain` catching a `URISyntaxException` that occurs sporadically, setting the status code, but not committing the change. Committing the status code change should fix the issue, but I haven't been able to figure out why a `URISyntaxException` ever occurs. For some reason, Repose determines the destination `URI` to be `//` which is not a valid `URI`, however looking at this test and the determination logic, `//` should never be the destination `URI`.